src/SoapCore/SoapEndpointMiddleware.cs

```diff
@@ -233,11 +233,11 @@ private async Task ProcessMeta(HttpContext httpContext, bool showDocumentation)
 		{
 			var baseUrl = httpContext.Request.Scheme + "://" + httpContext.Request.Host + httpContext.Request.PathBase + httpContext.Request.Path;
 			var xmlNamespaceManager = GetXmlNamespaceManager(null);
-			var bindingName = "BasicHttpBinding_" + _service.GeneralContract.Name;
+			var bindingName = _options.EncoderOptions[0].BindingName + "_" + _service.GeneralContract.Name;

```

Comment:
Since this variable will be set here, even if EncoderOptions.BindingName is null, I don't think the fallback, to use "BasicHttpBinding" as default name, works. 
Could you verify that the meta document is the same as before this PR, by default?

Reply:
Would you be happy to read from the EncoderOptions but if they are empty we use the deault value?
Something like this:

var bindingName = !string.IsNullOrWhiteSpace(_options.EncoderOptions[0].BindingName) ? _options.EncoderOptions[0].BindingName : "BasicHttpBinding_" + _service.GeneralContract.Name;

Reply:
Regarding the meta body if I understand this correctly, you are referring to the non web service specific fields. I can send you a copy to reconfirm. They are identical apart from now we have the soap12 tags instead of soap as expected and I have included a Policy with WsHttpBinding. When using a BasicBinding and Soap 1.1 the meta body is identical. If I did not understand your question, if you do not mind explaining a little bit more I am happy to check exactly what you need and get back to you.

Reply:
Yep, your suggested change for binding name seems correct. 
Just make sure that if one doesn't set any options the binding name etc. stays the same. Then I'm happy.

Reply:
I just updated the PR. I updated how we handle a default value if none is present. I updated the relavent tests. Checked the metadata for SOAP 1.1 before and after the PR and it seems correct. I removed couple of hard coded sections (PolicyReference, UsingAddressing) as I wrongly assumed that they were always needed.